### PR TITLE
graalvm-jdk@17: deprecate due to login requirement

### PR DESCRIPTION
**Pull request by krehel**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---

This is now walled behind a login screen.

graalvm

### CODE REVIEW

**Comment by daeho-ro** [comment]
Then maybe all of these too?
- https://github.com/search?q=repo:Homebrew/homebrew-cask%20oracle.com&type=code

**Comment by khipp** [comment]
@daeho-ro This change seems to only affect `graalvm-jdk@17` (see the [announcement](https://www.oracle.com/java/technologies/downloads/#graalvmjava17) regarding license changes and subscription).

File: Casks/g/graalvm-jdk@17.rb [hunk 1]
```diff
@@ -11,10 +11,7 @@
   desc "GraalVM from Oracle"
   homepage "https://www.graalvm.org/"
 
-  livecheck do
-    url "https://www.oracle.com/java/technologies/downloads/"
-    regex(/graalvm\s+for\s+jdk\s+(17(?:\.\d+)+)/i)
-  end
+  deprecate! date: "2024-10-20", because: :no_longer_meets_critieria
```
Comment (daeho-ro) [inline Casks/g/graalvm-jdk@17.rb:14]: Typo exist.
```suggestion
  deprecate! date: "2024-10-20", because: :no_longer_meets_criteria
```

**Comment by daeho-ro** [comment]
@khipp All right. But `oracle-jdk@17` should still be considered, the license changed it too.

**Review by khipp — APPROVED** [review]

**Pull request merged**